www/class/centreonTraps.class.php

```diff
@@ -120,27 +143,48 @@ public function setForm($form)
         $this->form = $form;
     }
 
+    /**
+     * Check if the OID has the good Format
+     *
+     * @param null|string $oid
+     * @return boolean
+     */
+    public function testOidFormat($oid = null)
+    {
+        if (preg_match('/^(\.([0-2]))|([0-2])((\.0)|(\.([1-9][0-9]*)))*$/', $oid) == true) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
     /**
      *
      * tests if trap already exists
      * @param $oid
      */
     public function testTrapExistence($oid = null)
     {
-        $id = null;
-        if (isset($this->form)) {
-            $id = $this->form->getSubmitValue('traps_id');
-        }
-        $query = "SELECT traps_oid, traps_id FROM traps WHERE traps_oid = '" . $this->db->escape($oid) . "'";
-        $res = $this->db->query($query);
-        $trap = $res->fetchRow();
+        if ($this->testOidFormat($oid) === true) {
+            $id = null;
+            if (isset($this->form)) {
+                $id = $this->form->getSubmitValue('traps_id');
+            }
+            $query = "SELECT traps_oid, traps_id FROM traps WHERE traps_oid = :oid ";
 
-        if ($res->rowCount() >= 1 && $trap["traps_id"] == $id) {
-            return true;
-        } elseif ($res->rowCount() >= 1 && $trap["traps_id"] != $id) {
-            return false;
-        } else {
-            return true;
+            $statement = $this->db->prepare($query);
+            $statement->bindValue(':oid', $oid, \PDO::PARAM_STR);
+            $statement->execute();
+
+            $trap = $statement->fetch(\PDO::FETCH_ASSOC);
+
+            if ($statement->rowCount() >= 1 && $trap["traps_id"] == $id) {
+                return true;
+            } elseif ($statement->rowCount() >= 1 && $trap["traps_id"] != $id) {
+                return false;
+            } else {
```

Comment:
This function is use as a callback to trigger or not an error in the form:

If no results are return from the query :
true => the trap don't exist so it can be created,

If results are return from the query:
false => The trap already exist, can't be created and an error will be trigger in the form